core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -66,43 +66,50 @@
  * file.
  */
 class DeleteFileIndex {
+  private static final DeleteFile[] NO_DELETES = new DeleteFile[0];
+
   private final Map<Integer, PartitionSpec> specsById;
   private final Map<Integer, Types.StructType> partitionTypeById;
   private final Map<Integer, ThreadLocal<StructLikeWrapper>> wrapperById;
-  private final long[] globalSeqs;
-  private final DeleteFile[] globalDeletes;
-  private final Map<Pair<Integer, StructLikeWrapper>, Pair<long[], DeleteFile[]>>
-      sortedDeletesByPartition;
+  private final DeleteFileGroup globalDeletes;
+  private final Map<Pair<Integer, StructLikeWrapper>, DeleteFileGroup> deletesByPartition;
+  private final boolean isEmpty;
 
   DeleteFileIndex(
-      Map<Integer, PartitionSpec> specsById,
+      Map<Integer, PartitionSpec> specs,
```

Comment:
Renamed `specsById` to stay on one line below.

Reply:
Do we need to care about style like this later?

Reply:
I feel like it is OK to do given that this PR already contains a few cosmetic changes.